src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -12770,7 +12770,7 @@ void Unit::SendPlaySpellVisual(uint32 id)
     WorldPacket data(SMSG_PLAY_SPELL_VISUAL, 8 + 4);
     data << uint64(GetGUID());
     data << uint32(id); // SpellVisualKit.dbc index
-    SendMessageToSet(&data, false);
+    SendMessageToSet(&data, true);
```

Comment:
does this change affect other Unit::SendPlaySpellVisual calls?

Reply:
No, i tested the ones that use this function and still works fine.

Reply:
It is required to make the player itself also see the visual. On 335 this was used for creatures only before and this is needed to expand this to the player itself as well. On 434 and probably master as well this is already changed.